mpc/coordinator/mpc_command.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+from typing import Optional, List
+from os.path import exists
+import os.path
+import subprocess
+
+
+class MPCCommand(object):
+    """
+    Wrapper around the 'mpc' utility.
+    """
+
+    def __init__(self, mpc_exe: Optional[str] = ""):
+        self.mpc_exe = mpc_exe or _default_mpc_bin()
+        assert exists(self.mpc_exe)
+
+    def linear_combination(
+            self,
+            powersoftau_file: str,
+            lagrange_file: str,
+            linear_comb_out_file: str,
+            degree: Optional[int] = None) -> bool:
+        args = ["linear-combination"]
+        args += ["--pot-degree", str(degree)] if degree else []
+        args += [powersoftau_file, lagrange_file, linear_comb_out_file]
+        return self._exec(args)
+
+    def phase2_begin(self, linear_comb_file: str, challenge_0_file: str) -> bool:
+        return self._exec(["phase2-begin", linear_comb_file, challenge_0_file])
+
+    def phase2_verify_contribution(
+            self,
+            orig_challenge: str,
+            response: str,
+            out_new_challenge: Optional[str] = None,
+            transcript: Optional[str] = None) -> bool:
+        args = ["phase2-verify-contribution"]
+        args += ["--new-challenge", out_new_challenge] \
+            if out_new_challenge else []
+        args += ["--transcript", transcript] if transcript else []
+        args += [orig_challenge, response]
+        return self._exec(args)
+
+    def phase2_verify_transcript(
+            self,
+            orig_challenge: str,
+            final_challenge: str,
+            transcript: str,
+            digest_file: Optional[str] = None) -> bool:
+        args = ["phase2-verify-transcript"]
+        args += ["--digest", digest_file] if digest_file else []
+        args += [orig_challenge, transcript, final_challenge]
+        return self._exec(args)
+
+    def phase2_contribute(
+            self,
+            challenge_file: str,
+            output_file: str,
+            digest_file: Optional[str] = None,
+            skip_user_input: bool = False) -> bool:
+        args = ["phase2-contribute", challenge_file, output_file]
+        args += ["--digest", digest_file] if digest_file else []
+        args += ["--skip-user-input"] if output_file else []
+        return self._exec(args)
+
+    def _exec(self, args: List[str]) -> bool:
+        cmd = [self.mpc_exe] + args
+        print(f"CMD: {' '.join(cmd)}")
+        comp = subprocess.run(cmd)
+        return 0 == comp.returncode
+
+
+def _default_mpc_bin() -> str:
+    return os.path.join(
+        os.path.dirname(__file__),
+        "..",
+        "..",
+        "build",
+        "src",
+        "mpc",
+        "mpc-test")
```

Comment:
What about making this path, part of the config file or the server?

Reply:
Originally I wasn't sure it was worth the extra complexity of per-phase config data at this stage, but I can add that (for the binary paths in both phases).  It does make sense.

Reply:
What about adding a function `get_phase_binaries` (or something like that) in the file https://github.com/clearmatics/zeth/blob/mpc-coordinator/mpc/coordinator/server_configuration.py, and wrap this function here by doing something like:
```python
def _default_mpc_bin() -> str:
    return os.path.join(
        get_phase_binaries(),
        "build",
        "src",
        "mpc",
        "mpc-test")
```
and likewise for the `poweroftau`?

I mean this is not a big deal to keep these hardcoded paths for now, but the config file of the server seems to be the right place to write such things (and this has the benefits to make the server more "configurable")